### PR TITLE
docs: improved docs

### PR DESCRIPTION
**Pull request by elianiva**
I met some people that get confused by nvim-compe docs so I tried to "improve" the docs so people can get started more easily. Some things might be missing like `allow_prefix_unmatch` because I don't know what this option does and maybe some other explanations.

I was thinking of adding how to use this with other autopairs such as `nvim-autopairs` because you have an example for `lexima` but I'm not sure if that's the scope of this project.

edit: I can also help converting the readme into vimdocs if you want :)

### CODE REVIEW

**Comment by hrsh7th** [comment]
Realy helpful. Thank you for @elianiva

File: README.md [hunk 1]
```diff
@@ -2,81 +2,134 @@
...
+let g:compe.source.path = v:true " path completion. default: `false`
+let g:compe.source.buffer = v:true " buffer completion. default: `false`
+let g:compe.source.vsnip = v:true " vsnip completion, make sure you have `vim-vsnip` installed. default: `false`
+let g:compe.source.nvim_lsp = v:true " nvim builtin LSP completion. default: `false`
+let g:compe.source.nvim_lua = v:true " nvim's lua stdlib completion. default: `false`
+let g:compe.source.your_awesome_source = {
+  " you can also override completion source configuration
+  " by setting it to a `table` instead of `boolean`
+  " see `Source Configuration` for more information
+  \}
```
Comment (hrsh7th) [inline README.md:76]: Sorry. I don't want to maintein same explanation with Lua example and VimL example both.
Can you remove this duplication?

Comment (elianiva) [reply README.md:76]: sure

Comment (hrsh7th) [reply README.md:76]: It would be good to add a separate section. Thank you.
I don't stick to comment for explanation.

**Comment by hrsh7th** [comment]
LGTM! Thank you very much.

If you have no more improvements in this PR, I will merge it.

**Comment by elianiva** [comment]
Yep, I think I'm done.

Probably want to work on vimdocs version of this so people can just use `:h nvim-compe` if they want to look the documentation, but that's a separate PR because that might take some time.

**Comment by hrsh7th** [comment]
Agree with you. I skipped vim's docs...

**Pull request merged**